roles/matrix-server/tasks/setup/setup_riot_web.yml

```diff
@@ -28,6 +28,7 @@
   with_items:
     - "riot.im.conf"
     - "config.json"
+    - "{{ matrix_riot_web_homepage_template }}"
```

Comment:
Thinking about it now..

A few lines above, we have such a thing:

```yaml
src: "{{ role_path }}/templates/riot-web/{{ item }}.j2"
```

.. meaning that it will try to look for the file within the `{{ role_path }}/templates/riot-web/` directory.

This means that people can't override `matrix_riot_web_homepage_template` in a useful way.
I imagine people may want to point it to another file (one that lives outside that directory), but it's not possible with the current code.

---------------------

It's probably best if the whole code is rewritten like this (note: untested!):

```yaml
- name: Ensure Matrix riot-web configured
  template:
    src: "{{ item.src }}"
    dest: "{{ matrix_riot_web_data_path }}/{{ item.name }}"
    mode: 0644
    owner: "{{ matrix_user_username }}"
    group: "{{ matrix_user_username }}"
  with_items:
    - {src: "{{ role_path }}/templates/riot-web/riot.im.conf.j2", name: "riot.im.conf"}
    - {src: "{{ role_path }}/templates/riot-web/config.json.j2", name: "config.json"}
    - {src: "{{ matrix_riot_web_homepage_template }}", name: "home.html"}
when: matrix_riot_web_enabled
```

With that, we can change the default for `matrix_riot_web_homepage_template` (in `roles/matrix-server/defaults/main.yml`) like this:

```yaml
matrix_riot_web_homepage_template: "{{ role_path }}/templates/riot-web/home.html"
```

.. and everything should work as expected.

Reply:
Makes sense, make an almost similar change.